### PR TITLE
Store the top-level and parent browsing contexts explicitly.

### PR DESCRIPTION
**Pull request by jgraham**
Previously the parent and top-level browsing contexts were computed
properties based on the current browsing context. This fails in the
case where the frame containing the current browsing context is
removed from the DOM tree since the browsing context is then discarded
meaning that trying to compute its parents will fail. In practice we
expect operations working with the top-level browsing context to
continue working in that case (and indeed implementations are
mostly/entirely not following the current spec), so this makes the
top-level and parent browsing contexts explictly stored properties of
the session that are updated at the point where we change browsing
contexts.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver/pull/1543.html" title="Last updated on Sep 8, 2020, 1:10 PM UTC (43f6e94)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver/1543/f0c7bdf...43f6e94.html" title="Last updated on Sep 8, 2020, 1:10 PM UTC (43f6e94)">Diff</a>

### CODE REVIEW

**Comment by whimboo** [comment]
Please note that appropriate WebDriver tests will be added/updated via https://bugzilla.mozilla.org/show_bug.cgi?id=1493108 once that PR has been approved.

File: index.html
Comment (whimboo) [inline index.html:3566]: Please note that the first check for `current browsing context` of the remote steps still cause an early abort of the command, which leads to not being able to switch to the parent context, if the current one is gone.

File: index.html
Comment (whimboo) [inline index.html:3480]: Please note that the check for the `current browsing context` above is still faulty and wouldn't allow switching to the top-level browsing context at all if the current one is gone, but returns early.

File: index.html
Comment (whimboo) [inline index.html:3493]: This should be a `no such frame` error, right? Or should it only be the used when the current browsing context is not the top-level one?

Comment (jgraham) [reply index.html:3493]: I think `no such frame` is only for the case where the specified frame can't be found, and `no such window` is always used for the case where the current browsing context can't be found.

File: index.html
Comment (mjzffr) [inline index.html:2167]: An extra "s" to remove?

> current parent browsing context, which is set to the parent of the current browsing context**s** when changing browsing contexts,

Comment (jgraham) [reply index.html]: Oops, thanks.

File: index.html
Comment (whimboo) [reply index.html:3493]: Ok, that makes sense. Otherwise also all the other commands would have to be changed too. And for those it doesn't make sense.

**Review by whimboo — APPROVED** [review]
From my perspective it looks fine, but it would be good to also get a review from @AutomatedTester and/or @shs96c.

**Review by AutomatedTester — APPROVED** [review]

**Pull request merged**
